src/common/marlin_server.cpp

```diff
@@ -1624,7 +1623,7 @@ void FSM_notifier::SendNotification() {
         progress = s_data.progress_max;
 
     // after first sent, progress can only rise
-    if ((s_data.last_progress_sent == uint8_t(-1)) || (progress > s_data.last_progress_sent)) {
+    if (!s_data.last_progress_sent.has_value() || (progress > s_data.last_progress_sent.value())) {
```

Comment:
According to the docs, this might also be covered by just the comparison. But here I don't mind the explicitness of this current solution.